src/go/spin-stub/main.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+const appName = "rancher-desktop"
+
+func main() {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		panic(fmt.Sprintf("failed to get user home directory: %s", err))
+	}
+	localAppData := os.Getenv("LOCALAPPDATA")
+	if localAppData == "" {
+		localAppData = filepath.Join(homeDir, "AppData", "Local")
+	}
+	err = os.Setenv("SPIN_DATA_DIR", filepath.Join(localAppData, appName, "spin"))
+	if err != nil {
+		panic(fmt.Sprintf("failed to set SPIN_DATA_DIR: %s", err))
+	}
+	exe, err := os.Executable()
+	if err != nil {
+		panic(fmt.Sprintf("failed to get executable path: %s", err))
```

Comment:
Just FYI, in Go, it's good practice to wrap errors with context rather than formatting them into strings; this preserves the original error type. In this case, after the `panic,` things may not recover, but it's still good to follow the idiomatic error handling.

Reply:
All these calls are to `panic()` which takes a single string argument, and it exits the process. There is no higher level caller that could unwrap an error.

So I will apply your wording changes but otherwise keep the code as-is.

Reply:
> ll these calls are to `panic()` which takes a single string argument

That is not correct, `panic()` can accept any type that satisfies the error interface or any other type, including strings, custom types, arbitrary values, or even nil interface. However, all of the things I mentioned above are suggestions, as I mentioned it's just a good practice to follow the go's idiomatic.

Reply:
Ok, sorry about that, I thought I had checked recently, and `panic()` only took a string. I've made the changes as suggested then.

I've replaced the final `panic()` with code to propagate the exit code instead; that seemed more like the right thing to do.

Reply:
Sorry, just realized that the final error check should stay as a fallback if the `err` is not an `ExitError`. But now it should be done.